File: usr/resources/Python/FibonacciTest.py
#!/usr/bin/env python
import unittest
import Fibonacci

class FibonacciTest(unittest.TestCase):
  f=open('Fibonacci_Python_results.csv','w') 
  @classmethod
  def setUpClass(cls):
    pass
    
  @classmethod
  def tearDownClass(cls):
    FibonacciTest.f.close()

  def setUp(self):
    pass

  def test_1(self):
    if Fibonacci.fibonacciChecker(1)==True:
      FibonacciTest.f.write("1,,1\n")
    else:
      FibonacciTest.f.write("0,Unsuccessful normal Fibonacci generation,1\n")
    self.assertEqual(Fibonacci.fibonacciChecker(1),True)

  def test_3(self):
    if Fibonacci.fibonacciChecker(3)==True:
      FibonacciTest.f.write("1,,2\n")
    else:
      FibonacciTest.f.write("0,Unsuccessful Fibonacci false positive detection,2\n")
    self.assertEqual(Fibonacci.fibonacciChecker(3),True)

  def test_6(self):
    if Fibonacci.fibonacciChecker(6)==True:
      FibonacciTest.f.write("1,,3\n")
    else:
      FibonacciTest.f.write("0,Large Fibonacci failue,3\n")
    self.assertEqual(Fibonacci.fibonacciChecker(6),True)

  def test_9(self):
    if Fibonacci.fibonacciChecker(9)==True:
      FibonacciTest.f.write("1,,4\n")
    else:
      FibonacciTest.f.write("0,Unsuccessful normal Fibonacci generation,4\n")
    self.assertEqual(Fibonacci.fibonacciChecker(9),True)

  def test_13(self):
    if Fibonacci.fibonacciChecker(13)==True:
      FibonacciTest.f.write("1,,5\n")
    else:
      FibonacciTest.f.write("0,Unsuccessful Fibonacci false positive detection,5\n")
    self.assertEqual(Fibonacci.fibonacciChecker(13),True)

  def test_15(self):
    if Fibonacci.fibonacciChecker(15)==False:
      FibonacciTest.f.write("1,,6\n")
    else:
      FibonacciTest.f.write("0,Large Fibonacci failue,6\n")
    self.assertEqual(Fibonacci.fibonacciChecker(15),False)

  def test_21(self):
    if Fibonacci.fibonacciChecker(21)==True:
      FibonacciTest.f.write("1,,7\n")
    else:
      FibonacciTest.f.write("0,Large Fibonacci failue,7\n")
    self.assertEqual(Fibonacci.fibonacciChecker(21),True)

if __name__ == '__main__':
  unittest.main()

File: usr/resources/Ruby/FibonacciTest.rb
#!/usr/bin/env ruby
require_relative 'Fibonacci'
require 'test/unit'

class FibonacciTest < Test::Unit::TestCase
      @@f = File.open("Fibonacci_Ruby_results.csv","w")
      
  def setup
      @@f.close unless @@f.closed?
      @@f = File.open("Fibonacci_Ruby_results.csv","a")
  end

  def teardown
    @@f.close
  end

  def testFibonacci1
    if fibonacciChecker(1)==true
      @@f.write("1,,1\n")   
    else
      @@f.write("0,Simple Fibonacci test failed,1\n")
    end
    assert_equal(fibonacciChecker(1),true)
  end

  def testFibonacci3
    if fibonacciChecker(3)==true
      @@f.write("1,,2\n")   
    else
      @@f.write("0,Simple Fibonacci falsification test failed,2\n")
    end
    assert_equal(fibonacciChecker(3),true)
  end

  def testFibonacci6
    if fibonacciChecker(6)==false
      @@f.write("1,,3\n")   
    else
      @@f.write("0,Large Fibonacci test failed Fibonacci test failed,3\n")
    end
    assert_equal(fibonacciChecker(6),false)
  end  


  def testFibonacci9
    if fibonacciChecker(9)==false
      @@f.write("1,,4\n")   
    else
      @@f.write("0,Simple Fibonacci test failed,4\n")
    end
    assert_equal(fibonacciChecker(9),false)
  end

  def testFibonacci13
    if fibonacciChecker(13)==false
      @@f.write("1,,5\n")   
    else
      @@f.write("0,Simple Fibonacci falsification test failed,5\n")
    end
    assert_equal(fibonacciChecker(13),false)
  end

  def testFibonacci15
    if fibonacciChecker(15)==false
      @@f.write("1,,6\n")   
    else
      @@f.write("0,Large Fibonacci test failed Fibonacci test failed,6\n")
    end
    assert_equal(fibonacciChecker(15),false)
  end  

  def testFibonacci21
    if fibonacciChecker(21)==true
      @@f.write("1,,7\n")   
    else
      @@f.write("0,Large Fibonacci test failed Fibonacci test failed,7\n")
    end
    assert_equal(fibonacciChecker(21),true)
  end  



end
